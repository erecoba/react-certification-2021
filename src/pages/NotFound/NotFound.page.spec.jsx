import React from 'react';
import { act, render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import NotFoundPage from './NotFound.page';

describe('Pages - NotFoundPage', () => {
  let history;

  beforeEach(() => {
    history = createMemoryHistory();
  });

  it('should render correctly NotFoundPage', async () => {
    let wrapper;

    await act(async () => {
      wrapper = render(
        <Router history={history}>
          <NotFoundPage />
        </Router>
      );
    });

    let fragment;
    await act(async () => {
      fragment = wrapper.asFragment(
        <Router history={history}>
          <NotFoundPage />
        </Router>
      );
    });

    expect(fragment).toMatchSnapshot();
  });
});
