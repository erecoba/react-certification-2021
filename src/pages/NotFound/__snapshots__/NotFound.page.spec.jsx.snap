// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages - NotFoundPage should render correctly NotFoundPage 1`] = `
<DocumentFragment>
  <section
    class="not-found"
  >
    <a
      class="home-link"
      href="/"
    >
      home
    </a>
    <img
      alt="page not found"
      src="404.gif"
    />
  </section>
</DocumentFragment>
`;
